src/platformio.ts

```diff
@@ -85,21 +98,124 @@ export class PlatformIO {
                 let includes = idedataJson.includes;
                 this.addIncludePathToFile(includes);
             } else {
-                this.showErrorToAddIncludePath();
+                if (!silent) {
+                    this.showErrorToAddIncludePath();
+                }
+
                 return;
             }
-            vscode.window.showInformationMessage("Add Include Path to c_cpp_properties.json successfully.");
+
+            if (!silent) {
+                vscode.window.showInformationMessage("Add Include Path to c_cpp_properties.json successfully.");
+            }
         } catch (ex) {
-            this.showErrorToAddIncludePath();
+            if (!silent) {
+                this.showErrorToAddIncludePath();
+            }
         } finally {
-            AppInsightsClient.sendEvent("addIncludePath");
+            if (!silent) {
+                AppInsightsClient.sendEvent("addIncludePath");
+            }
         }
     }
 
     public onDidCloseTerminal(closedTerminal: vscode.Terminal): void {
         this._pioTerminal.onDidCloseTerminal(closedTerminal);
     }
 
+    private configureAutoUpdateIncludes(): void {
+        const cppExtension = vscode.extensions.getExtension("ms-vscode.cpptools");
+        if (cppExtension) {
+            const autoUpdateIncludes = vscode.workspace.getConfiguration("platformio").get<boolean>("autoUpdateIncludes");
+            if (autoUpdateIncludes && !this._autoUpdateIncludesEnabled) {
+                this.buildFolderList();
+                const configFilePath = path.join(vscode.workspace.rootPath, ".vscode", "c_cpp_properties.json");
+                if (!Utility.fileExistsSync(configFilePath)) {
+                    this.addIncludePath(true);
+                }
+
+                // Watch .piolibdeps and lib folders
+                const fileSystemWatcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher("**/{.piolibdeps,lib}/**");
+                this._autoUpdateIncludesDisposables.push(fileSystemWatcher);
+                fileSystemWatcher.onDidCreate((e: vscode.Uri) => this.fileSystemDidCreate(e), this, this._autoUpdateIncludesDisposables);
+                fileSystemWatcher.onDidDelete((e: vscode.Uri) => this.fileSystemDidDelete(e), this, this._autoUpdateIncludesDisposables);

```

Comment:
The code can be made simpler if we don't care about deleted folders and I guess the C++ extension don't care if some include folders don't exist. 

I didn't realize that you only add paths to the c_cpp_properties.json file if it already exist. I think that is not a good idea and I will create an issue to explain why, but I think we should finish and merge this before fixing that.